configurations/assignmentEnvConfig.go

```diff
@@ -0,0 +1,101 @@
+package configurations
+
+import (
+	"assignment-exec/image-builder/constants"
+	"assignment-exec/image-builder/utilities/validation"
+	"bytes"
+	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+)
+
+type AssignmentEnvConfig struct {
+	BaseImage    string         `yaml:"baseImage"`
+	Dependencies LangDependency `yaml:"dependencies"`
+}
+
+func (config *AssignmentEnvConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	type tempAssignmentEnvConfig struct {
+		BaseImage    string         `yaml:"baseImage"`
+		Dependencies LangDependency `yaml:"dependencies"`
+	}
+	temp := &tempAssignmentEnvConfig{}
+
+	if err := unmarshal(temp); err != nil {
+		return errors.Wrap(err, "error in unmarshaling assignment environment configuration")
+	}
+
+	// Validate the configuration data.
+	err := validation.Validate("error in configuration",
```

Comment:
unmarshaling and validation can be propagated through nested structs.
Check this out that I just wrote - https://play.golang.org/p/pNssuApV2Kj

Reply:
Okay. Every nested struct is individually unmarshaled and validated.

Reply:
I don't see this change.